File: algorithms/sum_of_odds_in_triangle/SumOfOddsInTriangle.java
public class SumOfOddsInTriangle{
	
	/*
	 *Where n is the row number, calculate the first value in the row with: 
	 * 				(n * (n - 1)) + 1
	 * 
	 * After that increment that value in 2 same times as the row number. And adding each new value
	 * to the total variable;
	 * */
	public static int rowSumOddNumbers(int n){
		int total = 0;
		int counter = (n * (n - 1)) + 1;
		
		for(int i=0; i<n; i++) {
			total += counter;
			counter += 2;
		}		
		return total;
	}
}

File: algorithms/sum_of_odds_in_triangle/sum_of_odds_in_triangle.py
def rowSumOddNumbers(n):
    # The nth row starts with the number n^2-n+1 and then is an arithmetic 
    # sequence starting with that number and has lenght n, for a sum of
    # n^2-n+1 + n^2-n+3 + ... + n^2-n+2n-1 
    # = (n^2 + ... + n^2) - (n + ... + n) + (1 + 3 + ... + 2n-1)
    # = n^3 - n^2 + n^2 = n^3
    return n ** 3
